easybuild/easyblocks/generic/pythonpackage.py

```diff
@@ -502,9 +502,15 @@ def install_step(self):
         new_pythonpath = os.pathsep.join([x for x in abs_pylibdirs + [pythonpath] if x is not None])
         env.setvar('PYTHONPATH', new_pythonpath, verbose=False)
 
+        install_cmd = self.compose_install_command(self.installdir)
+        self.log.info("Installing %s v%s with: %s", self.name, self.version, install_cmd)
+
+        # run 'pip check' before trying to install package, if pip is used with --no-deps
+        if "pip install" in install_cmd and '--no-deps' in install_cmd:
+            run_cmd("pip check", log_all=True, log_ok=True, simple=False)
```

Comment:
this is checking the dependencies of installed packages, but the current one isn't installed yet

(it does work for the ones that are themselves used as dependencies later, but the "last" one isn't checked)

Reply:
This is checking whether all required dependencies of the package to-be-installed are in place; if not, `pip check` will fail.

So rather than checking for dependencies being downloaded during the installation, we're simply checking before the installation...

Reply:
After discussing this with @migueldiascosta it's clear that I misunderstood what `pip check` does.

It indeed checks whether all dependencies of *installed* packages are available, not of the package to-be-installed (which is how I understood this).

I'll rework this, and see if it's still useful.

Thanks for pointing this out @migueldiascosta!